### PR TITLE
various tools & libraries built with foss/2018b

### PR DESCRIPTION
**Pull request by fizwit**
adding libraries for foss-2018b

edit: depends on ~~#6656~~

### CODE REVIEW

**Comment by akesandgren** [comment]
This PR is targeting the wrong branch!
It should be against develop not master!

File: easybuild/easyconfigs/g/GDAL/GDAL-2.2.3-foss-2018b-Python-3.7.0.eb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+toolchainopts = {'usempi': True}
+
+source_urls = ['http://download.osgeo.org/gdal/%(version)s/']
+sources = [SOURCELOWER_TAR_XZ]
+checksums = ['a328d63d476b3653f5a25b5f7971e87a15cdf8860ab0729d4b1157ba988b8d0b']
+
+dependencies = [
+    ('Python', '3.7.0'),
+    ('netCDF', '4.6.1'),
+    ('expat', '2.2.5', '-GCCcore-7.3.0', True),
```
Comment (boegel) [inline easybuild/easyconfigs/g/GDAL/GDAL-2.2.3-foss-2018b-Python-3.7.0.eb]: @fizwit Why is this hardcoded to use `GCCcore`, there should be no need for this? Same below.

File: easybuild/easyconfigs/g/GDAL/GDAL-2.2.3-foss-2018b-Python-3.7.0.eb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    ('GEOS', '3.6.2', '-Python-%(pyver)s'),
+    ('SQLite', '3.24.0'), # GCCcore
+    ('libxml2', '2.9.8'), # GCCcore
+    ('libpng', '1.6.34'), # GCCcore
```
Comment (boegel) [inline easybuild/easyconfigs/g/GDAL/GDAL-2.2.3-foss-2018b-Python-3.7.0.eb]: No need for these `GCCcore` comments imho...

File: easybuild/easyconfigs/g/GDAL/GDAL-2.2.3-foss-2018b-Python-3.7.0.eb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    ('libjpeg-turbo', '2.0.0', '-GCCcore-7.3.0', True),
+    ('JasPer', '2.0.14'),
+    ('LibTIFF', '4.0.9', '-GCCcore-7.3.0', True),
+    ('zlib', '1.2.11'),
+    ('cURL', '7.60.0', '-GCCcore-7.3.0', True),
+    ('PCRE', '8.41'),
+    ('PROJ', '5.0.0'),
```
Comment (boegel) [inline easybuild/easyconfigs/g/GDAL/GDAL-2.2.3-foss-2018b-Python-3.7.0.eb]: `libgeotiff` dependency is missing (see `GDAL-2.2.3-foss-2018a-Python-3.6.4.eb`)

File: easybuild/easyconfigs/j/JasPer/JasPer-2.0.14-GCCcore-7.3.0.eb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    ('binutils', '2.30'),
+    ('CMake', '3.11.4'),
+]
+
+separate_build_dir = True
+
+configopts = '-DJAS_ENABLE_DOC=OFF '
+
+sanity_check_paths = {
+    'files': ['bin/jasper', 'lib64/libjasper.%s' % SHLIB_EXT],
```
Comment (boegel) [inline easybuild/easyconfigs/j/JasPer/JasPer-2.0.14-GCCcore-7.3.0.eb]: should be `('lib/libjasper.%s' % SHLIB_EXT, 'lib64/libjasper.%s' % SHLIB_EXT)`

File: easybuild/easyconfigs/l/LibTIFF/LibTIFF-4.0.9-GCCcore-7.3.0.eb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+toolchain = {'name': 'GCCcore', 'version': '7.3.0'}
+
+source_urls = [
+    'http://download.osgeo.org/libtiff/',
+    'ftp://ftp.remotesensing.org/pub/libtiff/',
+]
+sources = ['tiff-%(version)s.tar.gz']
+checksums = ['6e7bdeec2c310734e734d19aae3a71ebe37a4d842e0e23dbb1b8921c0026cfcd']
+
```
Comment (boegel) [inline easybuild/easyconfigs/l/LibTIFF/LibTIFF-4.0.9-GCCcore-7.3.0.eb]: @fizwit missing `binutils` build dependency

File: easybuild/easyconfigs/s/SWIG/SWIG-3.0.12-foss-2018b-Python-3.7.0.eb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+toolchain = {'name': 'foss', 'version': '2018b'}
+toolchainopts = {'pic': True}
+
+source_urls = [SOURCEFORGE_SOURCE]
+sources = [SOURCELOWER_TAR_GZ]
+checksums = ['7cf9f447ae7ed1c51722efc45e7f14418d15d7a1e143ac9f09a668999f4fc94d']
+
+dependencies = [
+    ('Python', '3.7.0'),
+    ('PCRE', '8.41', '-GCCcore-7.3.0', True),
```
Comment (boegel) [inline easybuild/easyconfigs/s/SWIG/SWIG-3.0.12-foss-2018b-Python-3.7.0.eb]: @fizwit no need to hardcode to `GCCcore`?

**Comment by boegel** [comment]
@fizwit Now #6656 and #6710 have been merged, the changes in https://github.com/fizwit/easybuild-easyconfigs/pull/3 should make this good to go...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 13 out of 13 (13 easyconfigs in this PR)
node2594.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/9c30a63dd722dbc77dd8ed0386fd3bac for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 13 out of 13 (13 easyconfigs in this PR)
node2003.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/28b6cdb4798c5df57352c3d79dd9e1e2 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @fizwit!

**Pull request merged**